### PR TITLE
 *: remove all the "errors.Trace()" function calls

### PR DESCRIPTION
**Pull request by zz-jason**
### What problem does this PR solve? 

Removing all the `errors.Trace()` function calls. 

### What is changed and how it works?

```sh
# 1. remove `errors.Trace`:
find . -name "*.go" | xargs sed -i '' -e 's/errors.Trace(\(.*\))/\1/g'

# 2. fix go imports:
goimports -w .
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by coocood** [comment]
Some errors returned by third-party packages may lose the trace?

**Comment by zhouqiang-cl** [comment]
The last fail of bench is due to some resource destroy failed(on clean stage). we are try to figure it out why and fix it

**Comment by gregwebs** [comment]
@coocood yes, we will end up losing error traces for errors not originating in TiDB code (that is using pingcap/errors).
Therefore I suggest we step back from this PR to doing a more thorough investigation. BTW, from looking at this PR I cannot tell if this benchmark is a good or bad result since it doesn't show a comparison to before the code change.

Note that we already optimized pingcap/errors to produce just one stack trace (no matter how many times `errors.Trace` is called).

The big question in my mind is: why are we going through error paths during this benchmark? Either TiDB is not reliable enough or we are using errors for things that we don't need to use them for. This is often the case in go programs for example when something is queried and not found. Often not found is a normal condition and does not need to be represented with an error interface.

File: bindinfo/handle.go [hunk 1]
```diff
@@ -17,11 +17,12 @@ import (
 	"bytes"
 	"context"
 	"fmt"
-	"go.uber.org/zap"
 	"sync"
 	"sync/atomic"
 	"time"
 
+	"go.uber.org/zap"
+
```
Comment (jackysp) [inline bindinfo/handle.go]: Why added an empty line?

Comment (jackysp) [inline bindinfo/handle.go]: Why added an empty line?

**Comment by zz-jason** [comment]
> Some errors returned by third-party packages or builtin packages may lose the trace?

Yeah. As recommended, we should use `errors.Wrap()` or `errors.AddStack()` to attach a stack trace. This is a huge PR, we should split it into small PRs to make sure every important error has a stack trace.

File: bindinfo/handle.go (hunk 1 above)
Comment (zz-jason) [reply bindinfo/handle.go]: this is modified by `goimports`

**Comment by gregwebs** [comment]
@zz-jason why are we going down error paths during the benchmark?

**Comment by zz-jason** [comment]
> @zz-jason why are we going down error paths during the benchmark?

I thought it may hurt the performance:
1. Some simple functions can't be inlined because of calling `errors.Trace()`
2. There is an `if` branch in `errors.Trace()`, which may hurt the CPU branch predicating.

**Comment by gregwebs** [comment]
> Some simple functions can't be inlined because of calling errors.Trace()

okay, that makes sense. We should probably be able to remove `errors.Trace` from simple functions.

> There is an if branch in errors.Trace(), which may hurt the CPU branch predicating.

But why are we invoking `errors.Trace()` a meaningful amount during the benchmark? What errors are we seeing? I suggest we should fix the errors or change the code to not use errors to represent normal conditions.

**Comment by zz-jason** [comment]
> But why are we invoking errors.Trace() a meaningful amount during the benchmark?

We use https://github.com/juju/errors instead of https://github.com/pkg/errors in the beginning. It requires us to call `errors.Trace()` to attach function name and file numbers to the stack trace. So there a lot of `errors.Trace()` in our codebase.

> What errors are we seeing?

Actually, we can see any possible error. And for an error, we may need the calling stack to find the root cause.

> I suggest we should fix the errors or change the code to not use errors to represent normal conditions.

Yeah, some functions don't need to return an error. But it's very difficult to find all these functions :joy:

**Comment by zz-jason** [comment]
Close due to outdated.

**Pull request closed without merging**